### ISSUE TITLE
some log-lik risks are nan in SL risk estimates

### ISSUE THREAD

**Issue by jlstiles** [benkeser/halplus#2]
Have you noticed trouble with log-lik risk estimates for the SL wrapper?  I see some risks are showing up as nan but SL is still giving a coefficient so the predictions and meta-learning in SL are working.

**Comment by benkeser** [comment]
I guess it could be giving predictions outside (0,1), which is causing the problem. Can you check that easily?

**Comment by jlstiles** [comment]
Yeah i can run the simulation again and keep track of this--good idea.  I
didn't think this was possible, however, since I thought lasso was just
logistic regression with an added constraint on the L1 norm of the
coefficient.  Should not all predictions be of form expit(\beta X) for a
binary outcome?

**Comment by jlstiles** [comment]
Yes, it is giving predictions greater than 1

```
devtools::install_github("jlstiles/Simulations")
library(hal9001)
library(Simulations)

set.seed(14819)
n=200
data =gendata(n, g0=g0_linear,Q0=Q0_trig1)
X=data
X$Y=NULL
Y=data$Y
X0=X1=X
X0$A=0
X1$A=1
newdata = rbind(X,X1,X0)

time = proc.time()
halres <- hal(Y = Y, newX = newdata, X = X, family = "binomial",
                                   verbose = FALSE, parallel = FALSE)
timehal = proc.time() - time

Qk = halres$pred[1:n]
riskhal = mean(Y*log(Qk)+(1-Y)*log(1-Qk))
```

**Comment by benkeser** [comment]
Yes, I'm not surprised that was at the root. I'd suggest just truncating predictions for now until we work out a better solution.

**Comment by jlstiles** [comment]
On a side note, is it possible to do log-lik loss in fitting hal?  Clearly we are using only sq error since predictions are going outside 0-1.  I thought glmnet had that option
